thirdeye/thirdeye-frontend/app/pods/components/rootcause-metrics/component.js

```diff
@@ -82,74 +91,91 @@ export default Component.extend({
-
-      const rows = filterPrefix(Object.keys(entities), 'thirdeye:metric:')
-        .map(urn => {
-          return {
-            urn,
-            links: links[urn],
-            isSelected: selectedUrns.has(urn),
-            label: toMetricLabel(urn, entities),
-            dataset: toMetricDataset(urn, entities),
-            score: humanizeScore(scores[urn]),
-            current: this._makeRecord(urn, 'current', entities, aggregates),
-            baseline: this._makeRecord(urn, 'baseline', entities, aggregates),
-            wo1w: this._makeRecord(urn, 'wo1w', entities, aggregates),
-            wo2w: this._makeRecord(urn, 'wo2w', entities, aggregates),
-            sortable_current: this._makeChange(urn, 'current', aggregates),
-            sortable_baseline: this._makeChange(urn, 'baseline', aggregates),
-            sortable_wo1w: this._makeChange(urn, 'wo1w', aggregates),
-            sortable_wo2w: this._makeChange(urn, 'wo2w', aggregates),
-            isExclusionWarning: isExclusionWarning(urn, entities)
-          };
-        });
+  metricsTableData: computed('selectedUrns', 'entities', 'aggregates', 'scores', 'links', 'compareMode', function () {
+    const { selectedUrns, entities, aggregates, scores, links, compareMode } = getProperties(
+      this,
+      'selectedUrns',
+      'entities',
+      'aggregates',
+      'scores',
+      'links',
+      'compareMode'
+    );
+    let rows;
 
-      return _.sortBy(rows, (row) => row.label);
+    if (compareMode === 'forecast') {
+      rows = filterPrefix(Object.keys(entities), 'thirdeye:metric:').map((urn) => {
+        return {
+          urn,
+          isSelected: selectedUrns.has(urn),
+          label: toMetricLabel(urn, entities),
+          dataset: toMetricDataset(urn, entities),
+          score: humanizeScore(scores[urn]),
+          current: this._makeRecord(urn, 'current', entities, aggregates),
+          baseline: this._makeRecord(urn, 'baseline', entities, aggregates),
+          yoy: this._makeRecord(urn, 'yoy', entities, aggregates),
+          interval: this._makeIntervalString(urn, aggregates),
+          inInterval: this._isInInterval(urn, aggregates),
+          sortable_current: this._makeChange(urn, 'current', aggregates),
+          sortable_baseline: this._makeChange(urn, 'baseline', aggregates),
+          sortable_yoy: this._makeChange(urn, 'yoy', aggregates),
+          isExclusionWarning: isExclusionWarning(urn, entities)
+        };
+      });
+    } else {
+      rows = filterPrefix(Object.keys(entities), 'thirdeye:metric:').map((urn) => {
+        return {
+          urn,
+          links: links[urn],
+          isSelected: selectedUrns.has(urn),
+          label: toMetricLabel(urn, entities),
+          dataset: toMetricDataset(urn, entities),
+          score: humanizeScore(scores[urn]),
+          current: this._makeRecord(urn, 'current', entities, aggregates),
+          baseline: this._makeRecord(urn, 'baseline', entities, aggregates),
```

Comment:
`urn`, `isSelected`, `label`, `dataset`, `score`, `current`, `baseline`, `isExclusionWarning` properties are same in both the `if` and `else` sections. Lets pull those out into a common object before the `if..else` and destructure it in that object in `return` in both `if..else` before proceeding to add unique properties in each.

Reply:
I'm not sure about this approach, since we are iterating through keys and building rows based on the keys.  Wouldn't this be adding a lot of computational steps?

Reply:
Actually yes, I misunderstood. I agree we should keep it they way it is.